java-desktop/src/main/java/com/couchbase/todo/model/Config.java

```diff
@@ -0,0 +1,103 @@
+package com.couchbase.todo.model;
+
+import java.util.Objects;
+
+import com.couchbase.lite.ConsoleLogger;
+import com.couchbase.lite.Database;
+import com.couchbase.lite.LogDomain;
+import com.couchbase.lite.LogLevel;
+import com.couchbase.todo.TodoApp;
+
+
+public final class Config {
+
+    private final boolean loggingEnabled;
+    private final boolean loginRequired;
+    private final int attempts;
+    private final int attemptsWaitTime;
+
+    private final String dbName;
+    private final String sgUri;
+
+    private final TodoApp.CR_MODE cr_mode;
+
+    public static class Builder {
+        private boolean loggingEnabled = TodoApp.LOG_ENABLED;
+        private boolean loginRequired = TodoApp.LOGIN_REQUIRED;
+
+        private String dbName = TodoApp.DB_DIR;
+        private String sgUri = TodoApp.SYNC_URL;
+
+        private int attempts;
+        private int attemptsWaitTime;
+
+        private TodoApp.CR_MODE cr_mode = TodoApp.SYNC_CR_MODE;
+
+        public Builder logging(boolean loggingEnabled) {
+            this.loggingEnabled = loggingEnabled;
+            return this;
+        }
+
+        public Builder login(boolean loginRequired) {
+            this.loginRequired = loginRequired;
+            return this;
+        }
+
+        public Builder dbName(String dbName) {
+            this.dbName = dbName;
+            return this;
+        }
+
+        public Builder sgUri(String sgUri) {
```

Comment:
for some reason, we all refer to those things as "SGW"s.  Probably better do that here, too.